### PR TITLE
Make `galgebra.lt.Mlt` somewhat usable again

### PR DESCRIPTION
**Pull request by eric-wieser**
This is best reviewed commit-by-commit. The summary is:

1. Revert the parts of fc598df56ac6573248a187d514262f6312b17f31 that started a refactor of `Mlt` that was unfinished. This fixes gh-171.
2. Fix some obvious typos in code that never worked, and add tests.
3. Make the attributes restored in 1. private

### CODE REVIEW

**Comment by eric-wieser** [comment]
Low coverage of the patch is from the last commit renaming a variable, I think it can be ignored.

**Comment by utensil** [comment]
Will review tomorrow, this is a complicated PR.

**Comment by eric-wieser** [comment]
Would it help if I left the last commit to a separate PR? I can split them before you review tomorrow if you think it would help.

**Comment by utensil** [comment]
When I actually sit down to read the code commit by commit, everything makes sense, good catch and a very illustrative test!

**Comment by eric-wieser** [comment]
Will rebase today

File: test/test_lt.py [hunk 1]
```diff
@@ -12,3 +14,38 @@ def test_lt_matrix(self):
...
+        assert T(a1, a2) == TA(a1, a2)
+
+        # for addition, argument slots are reused
+        assert (T + T)(a1, a2) == T(a1, a2) + T(a1, a2)
+        assert (T - T)(a1, a2) == T(a1, a2) - T(a1, a2)
+
+        # for multiplication, argument slots are chained
+        assert (T * T)(a1, a2, a3, a4) == TA(a1, a2) * T(a3, a4)
+        assert (T ^ T)(a1, a2, a3, a4) == TA(a1, a2) ^ T(a3, a4)
+        assert (T | T)(a1, a2, a3, a4) == TA(a1, a2) | T(a3, a4)
```
Comment (utensil) [inline test/test_lt.py:51]: More cases can be added:

```python
        alpha = st4d.mv('alpha', 'scalar')
        b  = st4d.mv('b', 'vector')

        # test other multilinear function properties
        assert T(alpha * a1, a2) == alpha * T(a1, a2)
        assert T(a1, alpha * a2) == alpha * T(a1, a2)
        assert T(a1 + b, a2) == T(a1, a2) + T(b, a2)
        assert T(a1, a2 + b) == T(a1, a2) + T(a1, b)
```

Tested locally to be working.

Comment (eric-wieser) [reply test/test_lt.py:51]: I'm not sure that test is interesting - we already proved that `T(x) == TA(x)`, so your test is just showing that `TA` is multilinear, which isn't a test of `Mlt` at all.

Comment (utensil) [reply test/test_lt.py:51]: Tests don't need to be interesting. They simply serve as verifying its dumb correctness.

These properties are specified in the docstring,  so it's the proper place to add them.

Comment (eric-wieser) [reply test/test_lt.py:51]: I'll add them in a follow-up, since then we can see if they affect coverage too (and CI is already done here anyway)

**Pull request merged**